File: modules/app-updater/android/src/main/java/org/tinycld/appupdater/AppUpdaterBundle.kt
package org.tinycld.appupdater

import android.content.Context

/**
 * Public bridge over [Store] so the (config-plugin-injected) RN host
 * `getJSBundleFile()` override can resolve a staged OTA bundle path without
 * reaching into the module's internals. Compiled into the AppUpdater library
 * alongside [AppUpdaterModule], so `MainApplication` can reference it by its
 * fully-qualified name `org.tinycld.appupdater.AppUpdaterBundle`.
 *
 * Thin wrapper — no behavior of its own; delegates to [Store.resolveBundlePath].
 */
object AppUpdaterBundle {
    /**
     * Returns the staged bundle file path to load, or `null` to fall back to the
     * embedded `index.android.bundle`. Safe to call once, early, from
     * `getJSBundleFile()`.
     */
    @JvmStatic
    fun stagedBundlePath(context: Context): String? = Store(context).resolveBundlePath()
}

File: modules/app-updater/ios/AppUpdaterBundle.swift
import Foundation

/// Public bridge over the internal `Store` so the (config-plugin-injected)
/// `AppDelegate.bundleURL()` can resolve a staged OTA bundle without importing
/// the module's internal `Store`. Compiled into the AppUpdater pod alongside
/// `AppUpdaterModule.swift` (the podspec globs `*.{swift,h,m}`), so the
/// AppDelegate can reference this public class by name.
///
/// Thin wrapper — no behavior of its own; delegates to `Store.shared`.
@objc public class AppUpdaterBundle: NSObject {
    /// Returns the staged bundle URL to load, or `nil` to fall back to the
    /// embedded `main.jsbundle`. Safe to call once, early, from `bundleURL()`.
    @objc public static func stagedBundleURL() -> URL? {
        Store.shared.resolveBundleURL()
    }
}
